engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -207,11 +211,15 @@ private[spel] class Typer(classLoader: ClassLoader) extends LazyLogging {
           }
       }
 
-      case e: StringLiteral => valid(Typed[String])
-
       case e: Ternary => withTypedChildren {
-        case condition :: onTrue :: onFalse :: Nil if condition.canBeSubclassOf(Typed[Boolean]) => Valid(Typed(onTrue, onFalse))
-        case _ => invalid("Invalid ternary operator")
+        case condition :: onTrue :: onFalse :: Nil =>
+          val superType = commonSupertypeFinder.commonSupertype(onTrue, onFalse)(NumberTypesPromotionStrategy.ToSupertype)
+          if (condition.canBeSubclassOf(Typed[Boolean]) && superType != Typed.empty) {
+            Valid(superType)
+          } else {
+            invalid(s"Invalid operands: $condition ? $onTrue : $onFalse")
```

Comment:
I was thinking about such case:
"#someVar.isCaseOne ? {fieldForOne: "value1"} : {defaultField: "value2"}
- and then encode it (e.g. in REST stuff)

In current implementation I would break, and as user I think I sometimes want to treat is as Any?
So 1. is difficult to do without coding, 2. is kind of brutal - this is just single case, I'd like strict type checking as a whole :)
Can't we just have Valid(Unknown) here?

Reply:
In your case you will get type `{fieldForOne: String, defaultField: String}`. Check `CommonSupertypeFinder`, comment "can't be sure intention of user - union of fields is more secure than intersection" and `unionOfFields` method ;-)

Reply:
Generally I've left union behaviour for unions, `Unknown`, `TypedObjectTypingResult`, intersection is for simple types and (feature toggle) for complex types

Reply:
ok, thanks